cupyx/distributed/array/_chunk.py

```diff
@@ -0,0 +1,228 @@
+import contextlib
+from itertools import chain
+from typing import Any, Iterator, Optional, Union
+
+import numpy
+
+from cupy._core.core import ndarray
+import cupy._creation.basic as _creation_basic
+import cupy._manipulation.dims as _manipulation_dims
+from cupy.cuda.device import Device
+from cupy.cuda.stream import Event
+from cupy.cuda.stream import Stream
+from cupy.cuda.stream import get_current_stream
+
+from cupyx.distributed.array import _modes
+from cupyx.distributed.array import _index_arith
+from cupyx.distributed.array import _data_transfer
+from cupyx.distributed.array._data_transfer import _Communicator
+
+
+class _ArrayPlaceholder:
+    # Mocks ndarray
```

Comment:
Self note, this is used when we are waiting chunks to arrive, to the current device